src/probability/kernel/disintegration.lean

```diff
@@ -0,0 +1,407 @@
+
+In order to obtain a disintegration for any standard Borel space, we use that these spaces embed
+measurably into `ℝ`: it then suffices to define a suitable kernel for `Ω = ℝ`. In the real case,
+we define a conditional kernel by taking for each `a : α` the measure associated to the Stieltjes
+function `cond_cdf ρ a` (the conditional cumulative distribution function).
+
+## Main definitions
+
+* `probability_theory.cond_kernel ρ : kernel α Ω`: conditional kernel described above.
+
+## Main statements
+
+* `probability_theory.kernel.const_eq_comp_prod`:
+  `kernel.const γ ρ = (kernel.const γ ρ.fst) ⊗ₖ (kernel.prod_mk_left (cond_kernel ρ) γ)`
+* `probability_theory.measure_eq_comp_prod`:
+  `ρ = ((kernel.const unit ρ.fst) ⊗ₖ (kernel.prod_mk_left (cond_kernel ρ) unit)) ()`
+* `probability_theory.lintegral_cond_kernel`:
+  `∫⁻ a, ∫⁻ ω, f (a, ω) ∂(cond_kernel ρ a) ∂ρ.fst = ∫⁻ x, f x ∂ρ`
+
+## TODO
+
+* The finite measure hypothesis can be weakened to σ-finite. The proof uses the finite case.
+* Beyond measures, we can find a disintegration for a kernel `α → Ω × Ω'` by applying the
+  construction used here for all `a : α` and showing additional measurability properties of the map
+  we obtain.
+
+-/
+
+open measure_theory set filter
+
+open_locale ennreal measure_theory topology probability_theory
+
+namespace probability_theory
+
+variables {α : Type*} {mα : measurable_space α}
+
+include mα
+
+section real
+
+/-! ### Disintegration of measures on `α × ℝ` -/
+
+/-- Conditional measure on the second space of the product given the value on the first, as a
+kernel. Use the more general `cond_kernel`. -/
+noncomputable
+def cond_kernel_real (ρ : measure (α × ℝ)) : kernel α ℝ :=
+{ val := λ a, (cond_cdf ρ a).measure,
+  property := measurable_measure_cond_cdf ρ }
+
+instance (ρ : measure (α × ℝ)) : is_markov_kernel (cond_kernel_real ρ) :=
+⟨λ a, by { rw cond_kernel_real, apply_instance, } ⟩
+
+lemma cond_kernel_real_Iic (ρ : measure (α × ℝ)) (a : α) (x : ℝ) :
+  cond_kernel_real ρ a (Iic x) = ennreal.of_real (cond_cdf ρ a x) :=
+measure_cond_cdf_Iic ρ a x
+
+lemma set_lintegral_cond_kernel_real_Iic (ρ : measure (α × ℝ)) [is_finite_measure ρ] (x : ℝ)
+  {s : set α} (hs : measurable_set s) :
+  ∫⁻ a in s, cond_kernel_real ρ a (Iic x) ∂ρ.fst = ρ (s ×ˢ Iic x) :=
+by { simp_rw [cond_kernel_real_Iic], exact set_lintegral_cond_cdf ρ x hs, }
+
+lemma set_lintegral_cond_kernel_real_univ (ρ : measure (α × ℝ))
+  {s : set α} (hs : measurable_set s) :
+  ∫⁻ a in s, cond_kernel_real ρ a univ ∂ρ.fst = ρ (s ×ˢ univ) :=
+by simp only [measure_univ, lintegral_const, measure.restrict_apply, measurable_set.univ,
+  univ_inter, one_mul, measure.fst_apply hs, ← prod_univ]
+
+lemma lintegral_cond_kernel_real_univ (ρ : measure (α × ℝ)) :
+  ∫⁻ a, cond_kernel_real ρ a univ ∂ρ.fst = ρ univ :=
+by rw [← set_lintegral_univ, set_lintegral_cond_kernel_real_univ ρ measurable_set.univ,
+  univ_prod_univ]
+
```

Comment:
maybe put `(ρ : measure (α × ℝ)) [is_finite_measure ρ]` as variables from this point on, to avoid repeating them in all lemmas?